ajax/dropdownSoftwareLicense.php

```diff
@@ -44,21 +44,20 @@
       $_POST['value'] = 0;
    }
 
-   $restrict = getEntitiesRestrictRequest(' AND', 'glpi_softwarelicenses', 'entities_id',
-                                          $_POST['entity_restrict'], true);
-
    // Make a select box
-   $query = "SELECT DISTINCT *
-             FROM `glpi_softwarelicenses`
-             WHERE `glpi_softwarelicenses`.`softwares_id` = '".$_POST['softwares_id']."'
-                   $restrict
-             ORDER BY `name`";
-   $result = $DB->query($query);
-   $number = $DB->numrows($result);
+   $iterator = $DB->request([
+      'DISTINCT'  => true,
+      'FROM'      => 'glpi_softwarelicenses',
+      'WHERE'     => [
+         'glpi_softwarelicenses.softwares_id'   => (int)$_POST['softwares_id']
```

Comment:
as we use the iterator and so prepared queries, is the cast needed ?

Reply:
Not really needed, but I do not think this is a problem as we're waiting for an integer